File: Python/JobArray/word_count/top5each.py
import os, sys
sys.path.append('../../')
from word_count_helpers import *

# Load the file names
dataDir = "../../../data/word_count/"
fnames = os.listdir(dataDir)

# Grab the arguments that are passed in
# This is the task id and number of tasks that can be used
# to determine which indices this process/task is assigned
my_task_id = int(sys.argv[1])
num_tasks = int(sys.argv[2])

# Assign indices to this process/task
my_fnames = fnames[my_task_id-1:len(fnames):num_tasks]

for fname in my_fnames:
    # Read in file and clean the text
    f = open(dataDir+fname, 'r', encoding='utf-8')
    text = cleantext(f.readlines())

    # Count number of times each word appears
    counts = countwords(text)

    # Sort and print the top 5 words with their counts
    top5 = sorted(counts, key=counts.get, reverse=True)[:5]
    str =''
    for k in top5:
        str=str + "%s: %s" % (k, counts[k]) + "; "
        
    print(str[:-2])

File: Julia/word_count/JobArray/top5each.jl
# Read in file of helper functions
# This file includes: cleantext, countwords
include("../word_count_helpers.jl")

# Load the file names
dataLoc = "../../../data/word_count/";
fnames = dataLoc.*readdir(dataLoc)

# Grab the argument that is passed in
# This is the index into fnames for this process
task_id = parse(Int,ARGS[1])
num_tasks = parse(Int,ARGS[2])

# Check to see if the index is valid (so the program exits cleanly if the wrong indices are passed)
for i in task_id+1:num_tasks:length(fnames)

    # Read in file and clean the text
    f = open(fnames[i])
    text=cleantext(readlines(f))

    # Count number of times each word appears
    wordCounts = countwords(text)

    # Sort and print the top 5 words with their counts
    rankedwords = sort(collect(wordCounts), by=x->x[2], rev=true)
    println(rankedwords[1:5,:])
    
end
